src/main/java/jcifs/smb/SmbSessionImpl.java

```diff
@@ -0,0 +1,1297 @@
+        byte[] token = new byte[0];
+        int state = 10;
+        final SmbComNegotiateResponse negoResp = (SmbComNegotiateResponse) trans.getNegotiateResponse();
+        boolean anonymous = this.credentials.isAnonymous();
+        do {
+            switch ( state ) {
+            case 10: /* NTLM */
+
+                if ( trans.hasCapability(SmbConstants.CAP_EXTENDED_SECURITY) ) {
+                    log.debug("Extended security negotiated");
+                    state = 20; /* NTLMSSP */
+                    break;
+                }
+                else if ( getContext().getConfig().isForceExtendedSecurity() ) {
+                    throw new SmbException("Server does not supported extended security");
+                }
+
+                log.debug("Performing legacy session setup");
+                if ( ! ( this.credentials instanceof NtlmPasswordAuthenticator ) ) {
+                    throw new SmbAuthException("Incompatible credentials");
+                }
+
+                NtlmPasswordAuthenticator npa = (NtlmPasswordAuthenticator) this.credentials;
+
+                request = new SmbComSessionSetupAndX(this.getContext(), negoResp, andx, getCredentials());
+                // if the connection already has a digest set up this needs to be used
+                request.setDigest(getDigest());
+                response = new SmbComSessionSetupAndXResponse(getContext().getConfig(), andxResponse);
+                response.setExtendedSecurity(false);
+
+                /*
+                 * Create SMB signature digest if necessary
+                 * Only the first SMB_COM_SESSION_SETUP_ANX with non-null or
+                 * blank password initializes signing.
+                 */
+                if ( !anonymous && isSignatureSetupRequired() ) {
+                    if ( isExternalAuth(getContext(), npa) ) {
+                        /*
+                         * preauthentication
+                         */
+                        try ( SmbSessionImpl smbSession = trans.getSmbSession(getContext().withDefaultCredentials());
+                              SmbTreeImpl t = smbSession.getSmbTree(getContext().getConfig().getLogonShare(), null) ) {
+                            t.treeConnect(null, null);
+                        }
+                    }
+                    else {
+                        log.debug("Initialize signing");
+                        byte[] signingKey = npa.getSigningKey(getContext(), negoResp.getServerData().encryptionKey);
+                        if ( signingKey == null ) {
+                            throw new SmbException("Need a signature key but the server did not provide one");
+                        }
+                        request.setDigest(new SMB1SigningDigest(signingKey, false));
+                    }
+                }
+
+                try {
+                    trans.send(request, response);
+                }
+                catch ( SmbAuthException sae ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 1050.